frontend/src/components/Table/Table.tsx

```diff
@@ -239,6 +239,17 @@ const validateProps = <T extends IIdentifiedObject, TFilter extends object = {}>
   if (props.hideToolbar === true && props.manualPagination === false) {
     throw Error('When hiding pagination toolbar manual pagination must be true.');
   }
+  if (props.manualPagination === false && (props.onRequestData !== undefined || props.pageIndex)) {
+    throw Error(
+      'When manual pagination is set to false do not pass in onRequestData or pageIndex.',
+    );
+  }
+  if (
+    (props.setSelectedRows || props.isSingleSelect || props.selectedRows) &&
+    !props.showSelectedRowCount
+  ) {
+    throw Error('Show selected row count required for table select logic');
+  }
```

Comment:
How about using props type composition and only exporting the types and not the composed type.

Something like:
```
type ITableProps = HiddenPagination | SelectableRows;
```
We would need to define the clusters of props that need to go together.

I know this would be more complicated to implement, but it would give some guidance to the consumer of this component

Reply:
I agree that the table component has grown in complexity a lot and is in dire need of some refactoring but I'd rather had this specific bug fixes now and deal with the rest as tech debt ticket to be prioritized. Thoughts?

Reply:
that's fair, expedient bug fixing sounds better. We can refactor it on dev rather than on test